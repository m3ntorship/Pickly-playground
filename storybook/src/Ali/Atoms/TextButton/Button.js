import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const {
    textButtonDisabled,
    textButtonsize,
    textButtonvariant,
    textButtonchildren,
  } = props;

  return (
    <button
      disabled={textButtonDisabled}
      className={`capitalize rounded-full focus:outline-none disabled:opacity-25 ${
        textButtonvariant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !textButtonDisabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !textButtonDisabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        textButtonsize === "Large"
          ? "text-md font-bold px-xl py-3.5 "
          : textButtonsize === "Medium"
          ? "text-base font-medium px-l py-xs "
          : "text-sm font-medium px-m py-1.5 "
      }
      ${textButtonDisabled && "cursor-not-allowed"}`}
    >
      {textButtonchildren}
    </button>
  );
};

Button.propTypes = {
  textButtonsize: PropTypes.oneOf(["Small", "Medium", "Large"]),
  textButtonvariant: PropTypes.oneOf(["Primary", "Secondary"]),
  textButtonDisabled: PropTypes.bool,
  textButtonchildren: PropTypes.string,
};
export default Button;
