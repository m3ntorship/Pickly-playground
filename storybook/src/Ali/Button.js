import React from "react";
import PropTypes from "prop-types";

const Button = (props) => {
  const { disabled, size, variant, children } = props;
  return (
    <button
      disabled={disabled}
      className={`capitalize rounded-full font-bold focus:outline-none disabled:opacity-25 ${
        variant === "Primary"
          ? `bg-primary-shd1  text-white ${
              !disabled && "hover:bg-primary"
            } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
          : `border text-primary-shd1 border-primary-shd1 ${
              !disabled && "hover:bg-grey-bg"
            } focus:ring-1 focus:ring-primary-shd1`
      } ${
        size === "Large"
          ? "text-md px-xl py-s "
          : size === "Medium"
          ? "text-base px-l py-xs "
          : "text-sm px-m py-xxs "
      }
      ${disabled ? "cursor-not-allowed" : ""}`}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["Small", "Medium", "Large"]),
  variant: PropTypes.oneOf(["Primary", "Secondary"]),
  disabled: PropTypes.bool,
  children: PropTypes.string,
};
export default Button;
