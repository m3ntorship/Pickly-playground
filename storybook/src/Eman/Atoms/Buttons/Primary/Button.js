import React from "react";
import PropTypes from "prop-types";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
  TYPE: {
    PRIMARY: "primary",
    SECONDARY: "secondary",
  },
  DISABLED: {
    ON: "on",
    OFF: "off",
  },
};

export const Button = ({ type, content, size, disabled }) => {
  return (
    <button
      className={`
  ${
    type === BUTTON_OPTIONS.TYPE.PRIMARY
      ? "rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
      : "rounded-full  text-primary  border-2 border-primary"
  },
        ${size === BUTTON_OPTIONS.SIZE.BIG && "px-xl py-s font-bold text-md"},
        ${
          size === BUTTON_OPTIONS.SIZE.MEDIUM &&
          "px-l py-xs font-semibold  text-base"
        },
        ${
          size === BUTTON_OPTIONS.SIZE.SMALL &&
          "px-m py-xxs font-semibold text-sm"
        },
        ${disabled === BUTTON_OPTIONS.DISABLED.ON && "opacity-25"}
        `}
    >
      {content}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  content: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.string,
};
