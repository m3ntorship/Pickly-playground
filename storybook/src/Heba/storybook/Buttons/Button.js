import React from "react";
import cn from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
};

export const Button = ({ primary, size }) => {
  return (
    <div className="flex justify-start items-center ">
      <button
        type="button"
        className={cn(
          "flex  items-center justify-center",
          primary
            ? "text-white bg-primary rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : " text-primary bg-white  border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25 fill-current",
          { "py-3 px-3 ": size === BUTTON_OPTIONS.SIZE.BIG },
          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },

          {
            "py-2 px-2": size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
        disabled
      >
        <svg
          className={cn(
            { "w-8 h-auto ": size === BUTTON_OPTIONS.SIZE.BIG },
            {
              "w-auto h-6": size === BUTTON_OPTIONS.SIZE.MEDIUM,
            },

            {
              "w-3 h-auto": size === BUTTON_OPTIONS.SIZE.SMALL,
            }
          )}
          viewBox="0 0 32 32"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.724 12.3906L16 18.1146L10.276 12.3906L8.39062 14.276L16 21.8853L23.6093 14.276L21.724 12.3906Z"
            className={cn(
              primary ? "text-white fill-current" : "text-primary fill-current"
            )}
          />
        </svg>
      </button>
    </div>
  );
};
