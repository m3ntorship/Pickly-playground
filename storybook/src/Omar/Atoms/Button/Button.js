import React from "react";
import PropTypes from "prop-types";

export const Button = (props) => {
  const {
    disabled = false,
    size = "sm",
    variant = "primary",
    content,
    onClick,
  } = props;

  return (
    <button
      data-testid="btn-case"
      disabled={disabled}
      onClick={onClick}
      className={`rounded-full border disabled:opacity-30 flex items-center ${
        variant === "primary"
          ? "bg-primary text-white hover:bg-primary-shd9 focus:outline-none focus:border-primary-shd9"
          : "bg-white text-primary border-primary border-solid hover:bg-grey-bg2 focus:border-2 focus:outline-none"
      } ${
        size === "lg"
          ? "text-md py-3.5 px-xl"
          : size === "sm"
          ? "text-sm py-1.5 px-m"
          : "text-base py-xs px-l"
      }`}
    >
      <span>{content}</span>
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(["primary", "secondary"]),
  className: PropTypes.string,
  content: PropTypes.string,
};
