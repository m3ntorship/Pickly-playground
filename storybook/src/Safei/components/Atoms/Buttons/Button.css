/* //using css for colors not available in config and used very few times. */
.primary:hover{
    background-color: #5F3DC4;
}
.primary:focus,.secondary:hover{
    border: 2px solid #5034A6;
}
.secondary:hover{
    background-color: #FAFAFC;
}
.text:hover{
    color:#0092C9;
}
.text:focus span{
    text-decoration: underline;
}