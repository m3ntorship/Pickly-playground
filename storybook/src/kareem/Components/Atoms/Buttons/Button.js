import React from "react";
import PropTypes from "prop-types";
import clsname from "classnames";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    LARGE: "large",
  },
};

export function Button({ primary, children, size }) {
  return (
    <div>
      <button
        type="button"
        className={clsname(
          primary
            ? "bg-primary text-white font-sans rounded-full focus:ring-2 focus:ring-primary-shd9 focus:outline-none disabled:opacity-25 mr -4"
            : "text-primary bg-white font-sans border border-primary rounded-full  focus:ring-2 focus:ring-primary-shd9 focus:outline-none mr-4 disabled:opacity-25",
          {
            "py-3.5 px-8 font-bold text-md": size === BUTTON_OPTIONS.SIZE.LARGE,
          },
          {
            "py-2 px-6 text-base font-medium":
              size === BUTTON_OPTIONS.SIZE.MEDIUM,
          },
          {
            "py-1.5 px-4 font-medium text-sm":
              size === BUTTON_OPTIONS.SIZE.SMALL,
          }
        )}
      >
        {children}
      </button>
    </div>
  );
}

Button.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  primary: PropTypes.bool,
  /**
   * What background color to use
   */
  children: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(["small", "medium", "large"]),
};

Button.defaultProps = {
  primary: true,
  size: "medium",
  children: this.props.children,
};

export default Button;
