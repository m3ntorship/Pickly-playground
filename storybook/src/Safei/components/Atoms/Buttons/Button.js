import React from "react";
import "./Button.css";

// const Button = ({ size, type, isActive, hasIcon, iconLocation }) => {
//   //hasIcon = ['true', false] >>> Boolean
//   //iconLocation = ['left', 'right', 'center']

// export default Button;
const Button = ({ size, type, disabled, text, hasIcon, iconDirection }) => {
  let opacity, _size, _type, iconOrder, marginSetting, iconColor;
  iconOrder = `flex-row`;
  if (!text) {
    text = "Button";
  } //make sure btn not empty if no text passed.
  ////size = ['lg', 'md', 'sm]:
  if (size == "lg") _size = "px-xl py-3.5 text-md font-bold ";
  else if (size == "md") _size = "px-l py-xs text-base font-medium";
  else {
    _size = "px-m py-1.5 font-medium text-sm";
  }
  //type = ['primary', 'secondary', 'text']
  if (type == "secondary") {
    _type = "secondary bg-white text-primary border border-primary";
  } else if (type == "text") {
    _type = "text text-accent focus-within";
  } else {
    _type = "primary bg-primary text-white border-2 border-primary";
  }
  //dsiabled?
  if (disabled) {
    opacity = "opacity-25 cursor-not-allowed";
  }

  if (!hasIcon) {
    return (
      <button
        onClick={() => {
          alert("clicked me?");
        }}
        className={` ${_type} ${_size}   rounded-full hover-state focus:outline-none focus-state flex items-center justify-center content-center ${opacity}`}
        disabled={disabled}
      >
        <span className="inline-block font-medium	">{text}</span>
      </button>
    );
  }
  if (hasIcon) {
    iconOrder = iconDirection == "left" ? "flex-row" : "flex-row-reverse";
    marginSetting = iconDirection == "left" ? "ml" : "mr";
    iconColor =
      type == "text" ? "#00A8E8" : type == "secondary" ? "#7048E8" : "white";
    return (
      <button
        className={` ${_type} ${_size}   rounded-full hover-state focus:outline-none focus-state flex ${iconOrder} items-center justify-center content-center ${opacity}`}
        disabled={disabled}
      >
        <span className="inline-block font-medium	">{text}</span>
        <span className="inline-block">
          <svg
            className={`h-3 w-3 mt-2.5 ${marginSetting}-3.5`}
            viewBox="0 0 12 8"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.293 0.29303L5.99997 4.58603L1.70697 0.29303L0.292969 1.70703L5.99997 7.41403L11.707 1.70703L10.293 0.29303Z"
              fill={`${iconColor}`}
            />
          </svg>
        </span>
      </button>
    );
  }
};

export default Button;
