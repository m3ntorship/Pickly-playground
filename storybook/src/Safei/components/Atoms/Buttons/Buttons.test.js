import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";

//Testing libr:
describe("test button comp on click functoinality", () => {
  it("should return 0 clicks when the button is clicekd while disabled and text passed", () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick} disabled={true} text={"Click Me"} />);
    userEvent.click(screen.getByText("Click Me"));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });
});

//Snapshot:
describe("render a primary lg button", () => {
  it("render primary button comp large size", () => {
    const tree = renderer
      .create(<Button size={"lg"} type={"primary"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render secondary button comp md size", () => {
    const tree = renderer
      .create(<Button size={"md"} type={"secondary"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render text button comp sm size", () => {
    const tree = renderer.create(<Button size={"sm"} type={"text"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("render primary small button comp when no props are passed", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
