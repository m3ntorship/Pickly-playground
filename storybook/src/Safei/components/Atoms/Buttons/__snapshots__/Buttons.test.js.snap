// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a primary lg button render primary button comp large size 1`] = `
<button
  className=" primary bg-primary text-white border-2 border-primary px-xl py-3.5 text-md font-bold    rounded-full hover-state focus:outline-none focus-state flex items-center justify-center content-center undefined"
  onClick={[Function]}
>
  <span
    className="inline-block font-medium	"
  >
    Button
  </span>
</button>
`;

exports[`render a primary lg button render primary small button comp when no props are passed 1`] = `
<button
  className=" primary bg-primary text-white border-2 border-primary px-m py-1.5 font-medium text-sm   rounded-full hover-state focus:outline-none focus-state flex items-center justify-center content-center undefined"
  onClick={[Function]}
>
  <span
    className="inline-block font-medium	"
  >
    Button
  </span>
</button>
`;

exports[`render a primary lg button render secondary button comp md size 1`] = `
<button
  className=" secondary bg-white text-primary border border-primary px-l py-xs text-base font-medium   rounded-full hover-state focus:outline-none focus-state flex items-center justify-center content-center undefined"
  onClick={[Function]}
>
  <span
    className="inline-block font-medium	"
  >
    Button
  </span>
</button>
`;

exports[`render a primary lg button render text button comp sm size 1`] = `
<button
  className=" text text-accent focus-within px-m py-1.5 font-medium text-sm   rounded-full hover-state focus:outline-none focus-state flex items-center justify-center content-center undefined"
  onClick={[Function]}
>
  <span
    className="inline-block font-medium	"
  >
    Button
  </span>
</button>
`;
